### ISSUE TITLE
Equipment upgrade planner

### ISSUE THREAD

**Issue by myangelkamikaze** [ElectronicObserverEN/ElectronicObserver#245]
- [x] Basic functionality
- [x] Plan ahead for equipment you don't have yet
- [x] Upgrade costs
- [ ] Upgrade days

### Basic functionality

It would be nice to have a tool where you could plan your equipment upgrades. From this tool, you could select one of your equipments and set the desired upgrade level (1~10 or conversion).

First, we need an equipment picker (EquipmentData picker to be exact). This picker should have filtering options similar to the ship filter in the event lock planner. The picker should probably be opened in a new window.
After you pick an equipment, it gets added into the upgrade planner where the equipment name, current upgrade level and the desired upgrade level are displayed.
The planner should subscribe to the equipment upgrade event and refresh the data accordingly.
The planner should also have the option to hide upgrades that were already finished.
The data saved in the database would be EquipmentData.MasterID (equipment drop ID), EquipmentDataMaster.EquipmentId (so that we can still display the name after a conversion), the desired upgrade level and if the planned item was finished (flag needed for the conversion case).
```cs
public class EquipmentUpgradePlanItemModel
{
    public int Id { get; set; } // primary key for the database, this should be auto increment and shouldn't get set manually
    public int? EquipmentMasterId { get; set; } // EquipmentData.MasterID
    public EquipmentId EquipmentId { get; set; } // EquipmentDataMaster.EquipmentId
    public UpgradeLevel DesiredUpgradeLevel { get; set; }
    public bool Finished { get; set; }
}
```

That would be the minimal basic functionality. There are a few possible improvements after the basic functionality is done.

### Plan ahead for equipment you don't have yet

With the basic functionality, you can only plan for the equipment you currently own. For example, if you want to make 2 maxed 46kai but don't own any yet, you would have to add 46 to the plan up to conversion, and only after converting you could plan for max 46kai.

For this, we would need another equipment picker (EquipmentDataMaster picker), which should have the same functionality as the other equipment picker.
The data format would have to be adjusted to include the EquipmentDataMaster.EquipmentId, with the EquipmentMasterId set to null.

The planner should then have a function that would check all the models with EquipmentMasterId null, and scan the player equipment list to see if any of the owned equipment could be assigned to this plan item. If such equipment exists, EquipmentMasterId gets set to the MasterID of that equipment.
Open question, how to make this function work in the expected way. Example, you have a maxed 46kai and want to make another one, if you now made a plan to max another 46kai with the previous one not in the planner, this function would assign your owned maxed 46kai to this plan and mark it as completed, which is not what you'd expect.

### Upgrade costs

It might be useful to know how much your upgrade planning will cost to execute (resources, special items and equipment).
The costs should be displayed for each individual upgrade plan item, and as a summary for all items.
For this, we'd need to get the upgrade requirement from somewhere, I think the Chinese wiki has something like that, but I have no idea about it.

### Upgrade days

It would also be useful to know on which days the planned equipment can be upgraded. The days could be displayed on the upgrade entries, similar to Akashi list. We could then also add filters to only show planned upgrades for specific days.

**Comment by Jebzou** [comment]
We could also have a view to display the planned upgrades
It would just be a read only grid (like the expedition checker)

It would be amazing combined with the "Upgrade costs" and "Upgrade days" extra tasks (filter on today's upgrades only, display the next upgrade cost)
